lib/iterable_basics.dart

```diff
@@ -154,6 +154,21 @@ extension IterableBasics<E> on Iterable<E> {
       ? 0
       : this.fold(0, (prev, element) => prev + addend(element));
 
+  /// Returns the average of all the values in this iterable, as defined by
+  /// [addend].
+  ///
+  /// Returns 0 if [this] is empty.
+  ///
+  /// Example:
+  /// ```dart
+  /// ['a', 'aa', 'aaa'].average((s) => s.length); // 2
+  /// ```
+  num average(num Function(E) addend) {
```

Comment:
nit: `addend` doesn't feel quite right here, since we're returning the average instead of a sum.

could you rename this parameter to `num Function(e) value` instead? Please update the docs + test names too if you can.

Reply:
Renamed the parameter and updated the doc block.